lightning/src/ln/channelmanager.rs

```diff
@@ -1723,70 +1771,75 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
-				let (to_remote_reserve_satoshis, to_self_reserve_satoshis) =
-					channel.get_holder_counterparty_selected_channel_reserve_satoshis();
-				res.push(ChannelDetails {
-					channel_id: (*channel_id).clone(),
-					counterparty: ChannelCounterparty {
-						node_id: channel.get_counterparty_node_id(),
-						features: InitFeatures::empty(),
-						unspendable_punishment_reserve: to_remote_reserve_satoshis,
-						forwarding_info: channel.counterparty_forwarding_info(),
-						// Ensures that we have actually received the `htlc_minimum_msat` value
-						// from the counterparty through the `OpenChannel` or `AcceptChannel`
-						// message (as they are always the first message from the counterparty).
-						// Else `Channel::get_counterparty_htlc_minimum_msat` could return the
-						// default `0` value set by `Channel::new_outbound`.
-						outbound_htlc_minimum_msat: if channel.have_received_message() {
-							Some(channel.get_counterparty_htlc_minimum_msat()) } else { None },
-						outbound_htlc_maximum_msat: channel.get_counterparty_htlc_maximum_msat(),
-					},
-					funding_txo: channel.get_funding_txo(),
-					// Note that accept_channel (or open_channel) is always the first message, so
-					// `have_received_message` indicates that type negotiation has completed.
-					channel_type: if channel.have_received_message() { Some(channel.get_channel_type().clone()) } else { None },
-					short_channel_id: channel.get_short_channel_id(),
-					outbound_scid_alias: if channel.is_usable() { Some(channel.outbound_scid_alias()) } else { None },
-					inbound_scid_alias: channel.latest_inbound_scid_alias(),
-					channel_value_satoshis: channel.get_value_satoshis(),
-					unspendable_punishment_reserve: to_self_reserve_satoshis,
-					balance_msat: balance.balance_msat,
-					inbound_capacity_msat: balance.inbound_capacity_msat,
-					outbound_capacity_msat: balance.outbound_capacity_msat,
-					next_outbound_htlc_limit_msat: balance.next_outbound_htlc_limit_msat,
-					user_channel_id: channel.get_user_id(),
-					confirmations_required: channel.minimum_depth(),
-					confirmations: Some(channel.get_funding_tx_confirmations(best_block_height)),
-					force_close_spend_delay: channel.get_counterparty_selected_contest_delay(),
-					is_outbound: channel.is_outbound(),
-					is_channel_ready: channel.is_usable(),
-					is_usable: channel.is_live(),
-					is_public: channel.should_announce(),
-					inbound_htlc_minimum_msat: Some(channel.get_holder_htlc_minimum_msat()),
-					inbound_htlc_maximum_msat: channel.get_holder_htlc_maximum_msat(),
-					config: Some(channel.config()),
-				});
-			}
-		}
-		let per_peer_state = self.per_peer_state.read().unwrap();
-		for chan in res.iter_mut() {
-			if let Some(peer_state) = per_peer_state.get(&chan.counterparty.node_id) {
-				chan.counterparty.features = peer_state.lock().unwrap().latest_features.clone();
+			let per_peer_state = self.per_peer_state.read().unwrap();
```

Comment:
Are we ok with only holding the `read` lock here, ie. the channels may change under the execution of this function, so the result might not be the actual channel state? Or would we rather want to take the `write` lock to ensure that the channels can't change?